File: MatrizLoca.java

import java.util.Scanner;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        long tiempoInicio = System.currentTimeMillis();
        Scanner sc = new Scanner(System.in);

        String[] registros = sc.nextLine().split(" ");
        int filas = Integer.parseInt(registros[0]);
        int cols = Integer.parseInt(registros[1]);
        int[][] matriz = new int[filas][cols];
        for (int i = 0; i < filas; i++) {
            String[] nums = sc.nextLine().split(" ");
            for (int f = 0; f < cols; f++) {
                matriz[i][f] = Integer.parseInt(nums[f]);
                if (f % 2 == 0) {
                    matriz[i][f] += 2;
                }
                if (i % 2 == 0) {
                    matriz[i][f] += 1;
                } else {
                    if (i % 2 != 0 && f % 2 != 0) {
                        matriz[i][f] -= 3;
                    }
                }
            }
        }
        for (int v = 0; v < matriz.length; v++) {
            for (int w = 0; w < matriz[v].length; w++) {
                if (w == matriz[v].length - 1) {
                    System.out.print(matriz[v][w]);
                } else {
                    System.out.print(matriz[v][w] + " ");
                }
            }
            System.out.println();
        }
        long tiempoFinal = System.currentTimeMillis();
        long tiempoTotal = tiempoFinal - tiempoInicio;

        System.out.println("El programa tardó " + tiempoTotal + " milisegundos en ejecutarse.");

    }
}

File: php/MatrizLoca.php
<?php
$tiempoInicio = microtime(true);
$stdin = fopen('php://stdin', 'r');

$registros = explode(" ", fgets($stdin));
$filas = intval($registros[0]);
$cols = intval($registros[1]);
$matriz = array();
for ($i = 0; $i < $filas; $i++) {
    $nums = explode(" ", fgets($stdin));
    for ($f = 0; $f < $cols; $f++) {
        $matriz[$i][$f] = intval($nums[$f]);
        if ($f % 2 == 0) {
            $matriz[$i][$f] += 2;
        }
        if ($i % 2 == 0) {
            $matriz[$i][$f] += 1;
        } else {
            if ($i % 2 != 0 && $f % 2 != 0) {
                $matriz[$i][$f] -= 3;
            }
        }
    }
}
for ($v = 0; $v < count($matriz); $v++) {
    for ($w = 0; $w < count($matriz[$v]); $w++) {
        if ($w == count($matriz[$v]) - 1) {
            echo $matriz[$v][$w];
        } else {
            echo $matriz[$v][$w] . " ";
        }
    }
    echo "\n";
}
$tiempoFinal = microtime(true);
$tiempoTotal = ($tiempoFinal - $tiempoInicio) * 1000;

echo "El programa tardó " . $tiempoTotal . " milisegundos en ejecutarse.\n";
fclose($stdin);
?>